scripts/safe_template_linter.py

```diff
@@ -1402,10 +1500,64 @@ def _check_mako_expressions(self, mako_template, has_page_default, results):
                 continue
 
             context = self._get_context(contexts, expression['start_index'])
-            self._check_filters(expression, context, has_page_default, results)
+            self._check_filters(mako_template, expression, context, has_page_default, results)
             self._check_deprecated_display_name(expression, results)
             self._check_html_and_text(expression, has_page_default, results)
 
+    def _check_javascript_contexts(self, mako_template, contexts, results):
+        """
+        Lint the JavaScript contexts for JavaScript violations inside a Mako
+        template.
+
+        Arguments:
+            mako_template: The contents of the Mako template.
+            contexts: A list of context dicts with 'type' and 'index'.
+            results: A list of results into which violations will be added.
+
+        Side effect:
+            Adds JavaScript violations to results.
+        """
+        javascript_start_index = -1
```

Comment:
@nedbat: That makes sense.  These are things a Java developer needs to get used to. :)